opentelemetry-api/opentelemetry/backend.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+The OpenTelemetry backend module defines and implements the API to access an
+implementation (backend) of the OpenTelemetry API.
+
+By default, if you call a getter function (e.g., :func:`tracer`) and the corresponding setter (e.g.,
+:func:`set_tracer`) wasn't called, you will get a default implementation, which is selected as follows:
+
+    1. If the environment variable OPENTELEMETRY_PYTHON_BACKEND_<getter-name>` (e.g.,
```

Comment:
Do we expect the user to specify exporter (and perhaps its parameters) and sampler in environment variables?

Reply:
The env var is mostly intended as an external override mechanism. In fact, I had Dynatrace in mind where the usual user experience is that you install the "OneAgent" on a machine and all supported technologies automagically report to Dynatrace (e.g., because through some mechanism the PYTHONPATH and OPENTELEMETRY_PYTHON_BACKEND_DEFAULT env vars for Python processes are automatically manipulated). Typical users should not need it.

I can think of three ways to configure the SDK:
1. Externally, via config files placed at certain places or environment variables (which could be a good way for things like PaaS products that come with an Tracing backend server and auto-configure OpenTelemetry to report to it.
2. In code, by creating instances of SDK objects manually and using `set_tracer`, etc. Or the SDK could provide a convenience function `setup_sdk(options...)` that calls `set_*` for you. However, this has the drawback that it overrides the environment variables which might be an indicator that the current semantics of them are not that good... Maybe the envvars should override the `set_*` functions?
3. The SDK could provide functions that configure defaults for created instances. That's a bit clunky though because it introduces additional mutable global state, but it avoids (works around?) the issue with the environment variables.

Reply:
In any case, I would not expect this loader module to take care of any of that, so I'm resolving this conversation.